crates/ruff_linter/resources/test/fixtures/pylint/boolean_chained_comparison.py

```diff
@@ -0,0 +1,121 @@
+# ------------------
+# less than examples
+# ------------------
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a < b and b < c:  # [boolean-chained-comparison]
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a < b and b <= c:  # [boolean-chained-comparison]
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a <= b and b < c:  # [boolean-chained-comparison]
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a <= b and b <= c:  # [boolean-chained-comparison]
+    pass
+
+# ---------------------
+# greater than examples
+# ---------------------
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a > b and b > c:  # [boolean-chained-comparison]
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a >= b and b > c:  # [boolean-chained-comparison]
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a > b and b >= c:  # [boolean-chained-comparison]
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a >= b and b >= c:  # [boolean-chained-comparison]
+    pass
+
+# -----------------------
+# multiple fixes examples
+# -----------------------
+
+a = int(input())
+b = int(input())
+c = int(input())
+d = int(input())
+if a < b and b < c and c < d:  # [boolean-chained-comparison]
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+d = int(input())
+e = int(input())
+if a < b and b < c and c < d and d < e:  # [boolean-chained-comparison]
+    pass
+
+# ------------
+# bad examples
+# ------------
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a > b or b > c: 
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a > b or b in (1, 2): 
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a < b and b > c: 
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a < b and b >= c: 
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a <= b and b > c: 
+    pass
+
+a = int(input())
+b = int(input())
+c = int(input())
+if a <= b and b >= c: 
+    pass
+
+# TODO: check if this example should be fixable or not
+a = int(input())
+b = int(input())
+c = int(input())
+if a > b and b < c: 
```

Comment:
This to me does seem like a case where it could be written as a single expression `a > b < c`, but I am not entirely sure.

Reply:
Possibly, but I'm not convinced that it improves readability.